spec/unit/coinbase/smart_contract_spec.rb

```diff
@@ -198,6 +198,100 @@
     end
   end
 
+  describe '.read' do
+    subject(:result) do
+      described_class.read(
+        network_id: network_id,
+        contract_address: contract_address,
+        method: method_name,
+        abi: abi,
+        args: args
+      )
+    end
+
+    let(:network_id) { :ethereum_mainnet }
+    let(:contract_address) { '0x1234567890123456789012345678901234567890' }
+    let(:method_name) { 'pureUint256' }
+    let(:args) { {} }
+    let(:abi) do
+      [{
+        'type' => 'function',
+        'name' => 'pureUint256',
+        'inputs' => [],
+        'outputs' => [{ 'name' => '', 'type' => 'uint256', 'internalType' => 'uint256' }],
+        'stateMutability' => 'pure'
+      }]
+    end
+
+    let(:smart_contracts_api) { instance_double(Coinbase::Client::SmartContractsApi) }
+    let(:api_response) do
+      Coinbase::Client::SolidityValue.new({
+                                            'type' => 'uint256',
+                                            'value' => '123456789'
+                                          })
+    end
+
+    before do
+      allow(Coinbase::Client::SmartContractsApi).to receive(:new).and_return(smart_contracts_api)
+
+      # Match against any ReadContractRequest with the expected method, args, and abi
+      allow(smart_contracts_api).to receive(:read_contract) do |network, address, request|
```

Comment:
Please follow any of the existing examples for these tests!

We do not do any assertions within `before` blocks or within an allow statement